homeassistant/components/person/__init__.py

```diff
@@ -75,217 +72,169 @@
-        """Delete person."""
-        if person_id not in self.storage_data:
-            raise ValueError("Invalid person specified.")
 
-        self.storage_data.pop(person_id)
-        self._async_schedule_save()
-        ent_reg = await self.hass.helpers.entity_registry.async_get_registry()
+async def async_setup(hass: HomeAssistantType, config: ConfigType):
+    """Set up the person component."""
+    entity_component = EntityComponent(_LOGGER, DOMAIN, hass)
+    id_manager = collection.IDManager()
+    yaml_collection = collection.YamlCollection(
+        logging.getLogger(f"{__name__}.yaml_collection"), id_manager
+    )
+    storage_collection = PersonStorageCollection(
+        PersonStore(hass, STORAGE_VERSION, STORAGE_KEY),
+        logging.getLogger(f"{__name__}.storage_collection"),
+        id_manager,
+        yaml_collection,
+    )
 
-        for entity in self.component.entities:
-            if entity.unique_id == person_id:
-                await entity.async_remove()
-                ent_reg.async_remove(entity.entity_id)
-                break
+    collection.attach_entity_component_collection(
+        entity_component, yaml_collection, lambda conf: Person(conf, False)
+    )
+    collection.attach_entity_component_collection(
+        entity_component, storage_collection, lambda conf: Person(conf, True)
+    )
 
-    @callback
-    def _async_schedule_save(self) -> None:
-        """Schedule saving the area registry."""
-        self.store.async_delay_save(self._data_to_save, SAVE_DELAY)
+    await yaml_collection.async_load(
+        await filter_yaml_data(hass, config.get(DOMAIN, []))
+    )
+    await storage_collection.async_load()
 
-    @callback
-    def _data_to_save(self) -> dict:
-        """Return data of area registry to store in a file."""
-        return {"persons": list(self.storage_data.values())}
+    hass.data[DOMAIN] = (yaml_collection, storage_collection)
 
-    async def _validate_user_id(self, user_id):
-        """Validate the used user_id."""
-        if await self.hass.auth.async_get_user(user_id) is None:
-            raise ValueError("User does not exist")
+    collection.StorageCollectionWebsocket(
+        storage_collection, DOMAIN, DOMAIN, CREATE_FIELDS, UPDATE_FIELDS
+    ).async_setup(hass, create_list=False)
 
-        if any(
-            person
-            for person in chain(self.storage_data.values(), self.config_data.values())
-            if person.get(CONF_USER_ID) == user_id
-        ):
-            raise ValueError("User already taken")
+    websocket_api.async_register_command(hass, ws_list_person)
 
-    async def _user_removed(self, event: Event):
-        """Handle event that a person is removed."""
+    async def _handle_user_removed(event: Event) -> None:
+        """Handle a user being removed."""
         user_id = event.data["user_id"]
-        for person in self.storage_data.values():
+        for person in storage_collection.async_items():
```

Comment:
Should we log if we failed to remove the requested user?

Reply:
No. It's only a clean up action to make sure that we do not reference users that don't exist. It's not a requirement that all persons are linked to a user (at this point in time).